### PR TITLE
Bio.GenBank: Remove error-hiding __gettattr__ method (#2528)

### PR DESCRIPTION
**Pull request by kblin**
Signed-off-by: Kai Blin <kblin@biosustain.dtu.dk>

This pull request addresses issue #2528

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by kblin** [comment]
I'd argue this is a good thing. I'll look into this. Funny the GenBank tests didn't fail for me. I couldn't run all of the tests because they got stuck somewhere in the NCBI tests section.

**Comment by kblin** [comment]
oh dear, there's so much code in there that clearly never worked...

**Comment by peterjc** [comment]
I agree making all the GenBank scanner/consumer interactions is a good thing. You are making me worry what else might be hiding behind that old shortcut.

**Comment by kblin** [comment]
```
diff --git a/Bio/GenBank/__init__.py b/Bio/GenBank/__init__.py
index 707ee3c30..a712d87be 100644
--- a/Bio/GenBank/__init__.py
+++ b/Bio/GenBank/__init__.py
@@ -1411,7 +1411,7 @@ class _FeatureConsumer(_BaseGenBankConsumer):
                     "Could not determine alphabet for seq_type %s" % self._seq_type
                 )
 
-        if not sequence and self.__expected_size:
+        if not sequence and self._expected_size:
             self.data.seq = UnknownSeq(self._expected_size, seq_alphabet)
         else:
             self.data.seq = Seq(sequence, seq_alphabet)
```

Just saying...

**Comment by peterjc** [comment]
Clearly I was lax about test coverage for the UnknownSeq corner case...

Fingers crossed it was just those two missing (or wrong) consumer methods.

**Comment by kblin** [comment]
There's a flake8 stupidity that I didn't catch, one more test coming up

**Comment by kblin** [comment]
oh, look, default action seems to be to close the PR, sorry. :)

**Comment by peterjc** [comment]
Did you say you had another test coming? Or are you done for now?

**Comment by kblin** [comment]
Sorry, I meant patch, not test. I'm done for now.

Sent from my phone, apologies for the short message.

On 28 Jan 2020 15:23, Peter Cock <notifications@github.com> wrote:

Did you say you had another test coming? Or are you done for now?

**Comment by peterjc** [comment]
Thanks Kai!
